File: api/src/models/AppUser.js
import mongoose from 'mongoose';

const appUserSchema = new mongoose.Schema(
  {
    employeeId: {
      type: String,
      trim: true,
      uppercase: true,
      sparse: true,
      unique: true
    },
    username: {
      type: String,
      required: true,
      unique: true,
      trim: true,
      lowercase: true
    },
    passwordHash: {
      type: String,
      select: false
    },
    fullName: {
      type: String,
      required: true,
      trim: true
    },
    role: {
      type: String,
      required: true,
      trim: true
    },
    roleId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'Role',
      default: null
    },
    department: {
      type: String,
      trim: true,
      default: ''
    },
    email: {
      type: String,
      trim: true,
      lowercase: true
    },
    activeStatus: {
      type: Boolean,
      default: true
    },
    canPrintBarcodeLabels: {
      type: Boolean,
      default: false
    }
  },
  { timestamps: true }
);

appUserSchema.index({ fullName: 'text', username: 'text', role: 'text' });

export const AppUser = mongoose.model('AppUser', appUserSchema);

File: ims-api-net/src/Ims.Domain/Masters/AppUser.cs
using Ims.Domain.Common;

namespace Ims.Domain.Masters;

public sealed class AppUser : EntityBase, IYearScoped
{
    public string YearDatabaseName { get; set; } = string.Empty;

    public string EmployeeId { get; set; } = string.Empty;

    public string Username { get; set; } = string.Empty;

    public string PasswordHash { get; set; } = string.Empty;

    public string FullName { get; set; } = string.Empty;

    public string Role { get; set; } = string.Empty;

    public string? RoleId { get; set; }

    public string? Department { get; set; }

    public string? Email { get; set; }

    public bool ActiveStatus { get; set; } = true;

    public bool CanPrintBarcodeLabels { get; set; }
}
